test/unit/module/real/core/convert/double/convert.hpp

```diff
@@ -40,3 +40,22 @@ TTS_CASE_TPL("Check eve::convert behavior", EVE_TYPE)
   TTS_EQUAL(eve::convert((T(42.69))      , eve::as<double>()), static_cast<target_t>(v_t(42.69)) );
   TTS_EQUAL(eve::convert(eve::valmax(eve::as<T>()), eve::as<double>()), static_cast<target_t>(eve::valmax(eve::as<v_t>())) );
 }
+
+TTS_CASE_TPL("Check eve::convert logical behavior", EVE_TYPE)
+{
+#if defined(EVE_SIMD_TESTS)
+  using target_t = eve::logical<eve::wide<double, eve::fixed<EVE_CARDINAL>>>;
+#else
+  using target_t = eve::logical<double>;
+#endif
+
+  TTS_EQUAL(eve::convert(eve::logical<T>(true)  , eve::as<eve::logical<double>>()), target_t(true) );
+  TTS_EQUAL(eve::convert(eve::logical<T>(false) , eve::as<eve::logical<double>>()), target_t(false) );
+
+#if defined(EVE_SIMD_TESTS)
+  eve::logical<T> mixed( [](auto i, auto) { return i%2 == 0;});
+  target_t        ref( [](auto i, auto) { return i%2 == 0;});
+
+  TTS_EQUAL(eve::convert(mixed, eve::as<eve::logical<double>>()), ref );
+#endif
+}
```

Comment:
Just do two tests?

Reply:
Same. This test is run in both mode scalar and simd so I'll end up havign to protect the lambda constructor